### ISSUE TITLE
Does not work with wildcard syntax.

### ISSUE THREAD

**Issue by breerly** [orchestrate-io/dvr#8]
Trying to run tests with `dvr.record` flag and wildcard syntax:

```
$ go test -dvr.record github.com/breerly/myproject/...
?       github.com/breerly/myproject  [no test files]
```

When I remove the `dvr.record` flag, it works:

```
$ go test github.com/breerly/myproject...
ok      github.com/breerly/myproject/package1  0.579s
ok      github.com/breerly/myproject/package2     0.006s
```

Any help would be appreciated.

**Comment by liquidgecka** [comment]
Hrm.. what version of go are you using?

Also, try this:
```go test github.com/breerly/myproject/package1 -dvr.record ```

For some reason it seems that go is picky about parameter placement now..

**Comment by breerly** [comment]
Still no luck.

```
$ go version
go version go1.4.2 darwin/amd64
```

Moving the parameter to the end seems not to do anything useful.

```
$ go test github.com/breerly/myproject/package1 -dvr.record
flag provided but not defined: -dvr.record
Usage of /var/folders/p_/v79tzhk95dngdb92680m8vwm0000gn/T/go-build130905158/github.com/breerly/myproject/package1/_test/package1.test:
  -test.bench="": regular expression to select benchmarks to run
  -test.benchmem=false: print memory allocations for benchmarks
  -test.benchtime=1s: approximate run time for each benchmark
  -test.blockprofile="": write a goroutine blocking profile to the named file after execution
  -test.blockprofilerate=1: if >= 0, calls runtime.SetBlockProfileRate()
  -test.coverprofile="": write a coverage profile to the named file after execution
  -test.cpu="": comma-separated list of number of CPUs to use for each test
  -test.cpuprofile="": write a cpu profile to the named file during execution
  -test.memprofile="": write a memory profile to the named file after execution
  -test.memprofilerate=0: if >=0, sets runtime.MemProfileRate
  -test.outputdir="": directory in which to write profiles
  -test.parallel=1: maximum test parallelism
  -test.run="": regular expression to select tests and examples to run
  -test.short=false: run smaller test suite to save time
  -test.timeout=0: if positive, sets an aggregate time limit for all tests
  -test.v=false: verbose: print additional output
FAIL	github.com/breerly/myproject/package1	0.006s
```

**Comment by liquidgecka** [comment]
Hrm.. Are you sure that package tests files import dvr somewhere? I have the same version/os setup to test with and its working for me. The dvr package needs to be imported in at least one _test.go file in that directory otherwise the flag won't get added.

I tested it out with a simple package called "foo" which had this file in it: http://play.golang.org/p/ZJTYY8Fcp0

I can run that test using this command: ```go test -v github.com/liquidgecka/foo -dvr.record``` and it runs the internal tests and all. Try that to make sure the basic setup is the same. If so we can debug from there.

**Comment by breerly** [comment]
I think you'll want a test-case that uses the go tool wildcard syntax.

Here are my findings.

```
$ go test github.com/breerly/myproject/package1 -dvr.record
```

Works only when:

1. There is a file called `myproject/package1/testdata/archive.dvr`.
2. There is the proper dvr import statement in EVERY file in `package1`.

This is exasperated when using the `...` syntax:

1. Every package must have a `testdata/archive.dvr` file in it.
2. Every test in every package must have the import dvr statement in it.

A way around this seems to be:

```
go test github.com/breerly/myproject/... -dvr.file=/Users/breerly/go/src/github.com/breerly/myproject/archive.dvr -dvr.record
```

Unfortunately you have to use the absolute path to the dvr file since the wildcard syntax runs go test from each packages directory.

I think a good default would be looking in the project root for the dvr file when the `...` syntax is given.

**Comment by liquidgecka** [comment]
I don't think that every test file in each package has to have it, but yea you are right that every test package ends up needing it.. This gets annoying with recursion as well since it requires _everything_ to use it which is rarely what is expected.

I could easily add a option to do this via an environment variable so the testing flags won't get in the way if you want.. that would work across recursion as well.

Its also interesting to me that they change the behavior of the recursive tests because it used to run them from the same root dir which broke a TON of things I wrote in the early go 1.0/1.1 days. The biggest issue is that as a library I am not given access to any of the arguments of the go test side of things. So I don't know that its recursive (./...) or some such. I only get to see the view from inside of each recursive test being executed. =/

**Comment by dnaeon** [comment]
+1

Getting the same errors as in https://github.com/orchestrate-io/dvr/issues/8#issuecomment-112821343 when I try to run the tests.

Some of my tests are importing `dvr`, and some are not, and apparently if you don't import `dvr` in *all* tests you get errors.

**Comment by liquidgecka** [comment]
@dnaeon do you mean that you need to import it in all test files in a single package, or all test files in all packages used with ... ? Go test is seriously limited when running in ... mode because it basically just spins up a new test for each package and runs them.. I have tried getting around that every way I could but its basically the way that go devs want it (they don't want complex tests from what I can tell).

If its within a single package then that is odd.. The argument should exist for that specific test output (each package is compiled to a different binary).. If I ever get some free time I will make the args environment variables as well so its possible to run with ./... in testing
